src/lib.rs

```diff
@@ -12,9 +14,10 @@ use std::env;
 
 pub mod schema;
 pub mod models;
+pub mod errors;
 
 pub fn establish_connection() -> PgConnection {
     dotenv().ok();
-    let database_url = env::var("DATABASE_URL").expect("DATABASE_URL must be set");
-    PgConnection::establish(&database_url).expect(&format!("Error connecting to {}", database_url))
+    let database_url = env::var("DATABASE_URL").chain_err("DATABASE_URL must be set");
+    PgConnection::establish(&database_url).chain_err(&format!("Error connecting to {}", database_url))
```

Comment:
Should use `chain_err(|| &format!(...))` instead. This will cause an extra allocation every time this functions is called.

Reply:
Rustlang String Processing 101 👍